src/main.c

```diff
@@ -1,125 +1,130 @@
 
 void SysTick_Handler(void) {
-    msTicks++;
+  msTicks++;
 }
 
 /**
  * @details reads incoming CAN messages and prints information to the terminal
  */
 void Process_CAN_Inputs(void) {	
-	CCAN_MSG_OBJ_T rx_msg;
-	uint32_t ret = CAN_Receive(&rx_msg);
-
-        if (ret == NO_CAN_ERROR) {
-		Board_Println("Received CAN message.");
-
-		Board_Print("ID: ");
-		Board_Println_Int(rx_msg.mode_id, 10);
-
-		Board_Println("Data: ");
-		Board_Print("data[0]: ");
-		Board_Println_Int(rx_msg.data[0], 2);
-		Board_Print("data[1]: ");
-		Board_Println_Int(rx_msg.data[1], 2);
-		Board_Print("data[2]: ");
-		Board_Println_Int(rx_msg.data[2], 2);
-		Board_Print("data[3]: ");
-		Board_Println_Int(rx_msg.data[3], 2);
-		Board_Print("data[4]: ");
-		Board_Println_Int(rx_msg.data[4], 2);
-		Board_Print("data[5]: ");
-		Board_Println_Int(rx_msg.data[5], 2);
-		Board_Print("data[6]: ");
-		Board_Println_Int(rx_msg.data[6], 2);
-		Board_Print("data[7]: ");
-		Board_Println_Int(rx_msg.data[7], 2);
-
-	}
+  CCAN_MSG_OBJ_T rx_msg;
+  uint32_t ret = CAN_Receive(&rx_msg);
+
+  if (ret == NO_RX_CAN_MESSAGE) {
+    // No message received.
+    // TODO really need a way to distinguish no message available
+    // from an error while receiving messages...
+  } else if (ret == NO_CAN_ERROR) {
+    Serial_Print("CAN_rcv, id=");
+    Serial_PrintlnNumber(rx_msg.mode_id, 10);
+  } else {
+    // Should never happen
+    Serial_Println("Unexpected CAN error state!!");
+  }
+}
+
+void sendADCMessage() {
+  uint8_t steering_val = ADC_Read_Byte(STEERING_CHANNEL);
+  uint8_t accel_1_val = ADC_Read_Byte(ACCEL_1_CHANNEL);
+  uint8_t accel_2_val = ADC_Read_Byte(ACCEL_2_CHANNEL);
+  uint8_t brake_1_val = ADC_Read_Byte(BRAKE_1_CHANNEL);
+  uint8_t brake_2_val = ADC_Read_Byte(BRAKE_2_CHANNEL);
+
+  /*
+  Serial_Print("s: ");
+  Serial_PrintNumber(steering_val, 10);
+  Serial_Print(", a1: ");
+  Serial_PrintNumber(accel_1_val, 10);
+  Serial_Print(", a2: ");
+  Serial_PrintNumber(accel_2_val, 10);
+  Serial_Print(", b1: ");
+  Serial_PrintNumber(brake_1_val, 10);
+  Serial_Print(", b2: ");
+  Serial_PrintlnNumber(brake_2_val, 10);
+  */
+
+  const uint32_t can_out_id = FRONT_CAN_NODE_ANALOG_SENSORS__id;
+  const uint8_t steering_idx = __FRONT_CAN_NODE_ANALOG_SENSORS__STEERING__start >> 3;
+  const uint8_t accel_1_idx = __FRONT_CAN_NODE_ANALOG_SENSORS__RIGHT_ACCEL__start >> 3;
+  const uint8_t accel_2_idx = __FRONT_CAN_NODE_ANALOG_SENSORS__LEFT_ACCEL__start >> 3;
+  const uint8_t brake_1_idx = __FRONT_CAN_NODE_ANALOG_SENSORS__FRONT_BRAKE__start >> 3;
+  const uint8_t brake_2_idx = __FRONT_CAN_NODE_ANALOG_SENSORS__REAR_BRAKE__start >> 3;
+
+  const uint8_t can_out_bytes = 5;
+  uint8_t data[can_out_bytes];
+
+  data[steering_idx] = steering_val;
+  data[accel_1_idx] = accel_1_val;
+  data[accel_2_idx] = accel_2_val;
+  data[brake_1_idx] = brake_1_val;
+  data[brake_2_idx] = brake_2_val;
+
+  CAN_Transmit(can_out_id, data, can_out_bytes);
```

Comment:
Lol keep forgetting how sneaky C can be. Yeah if you can update it that'd be great, or I could do it if you're hosed. I assume the 10 possible errors are listed in the LPC manual? (The third link you put is a dupe of the first).